cmd/stats/stats.go

```diff
@@ -2,28 +2,218 @@ package stats
+			return todayAtMidnight, fmt.Errorf("invalid date: %s hasn't happend yet!", arg)
+		}
+
+		return argTime, nil
+	}
+
+	switch unit {
+	case 'H', 'h':
+		return time.Date(
+			now.Year(), now.Month(), now.Day(),
+			now.Hour()-amount, now.Minute(), now.Second(), now.Nanosecond(),
+			now.Location()), nil
+	case 'D', 'd':
+		return todayAtMidnight.AddDate(0, 0, -1*amount), nil
+	case 'W', 'w':
+		return todayAtMidnight.AddDate(0, 0, amount*-7), nil
+	case 'M', 'm':
+		return todayAtMidnight.AddDate(0, -1*amount, 0), nil
+	case 'Y', 'y':
+		return todayAtMidnight.AddDate(-1*amount, 0, 0), nil
+	default:
+		return todayAtMidnight, fmt.Errorf("invalid date format %s\nsee \"sweet stats --help\" for more details", arg)
+	}
+}
+
+// Converts the flags assigned to the stats command into an SQLite query,
+// retrieving all entries from the database that match the query.
+func argsToQuery(cmd *cobra.Command, now time.Time) (string, error) {
+	filters := []string{}
+	name := cmd.Flag("name").Value.String()
+	lang := cmd.Flag("lang").Value.String()
+
+	if name != "" && lang != "" {
+		return "", fmt.Errorf("both name and lang provided (please pick one of them!)")
+	} else if lang != "" {
+		filters = append(filters, fmt.Sprintf("lang='%s'", lang))
+	} else if name != "" {
+		nameFilter := fmt.Sprintf("name like '%s'", name)
+		nameFilter = strings.Replace(nameFilter, "*", "%", -1)
+		filters = append(filters, nameFilter)
+	}
+
+	end := cmd.Flag("end").Value.String()
+	since := cmd.Flag("since").Value.String()
+	start := cmd.Flag("start").Value.String()
+
+	if end != "" && since == "" && start == "" {
+		return "", fmt.Errorf("must define start if end is provided")
+	}
+
+	if since != "" && start != "" {
+		fmt.Printf("both `--since` and `--start` variables provided (you only need one of them!)")
+	} else if since != "" && start == "" {
+		start = since
+	}
+
+	startTime, _ := parseDateFromArg(false, start, now)
+	filters = append(filters, fmt.Sprintf("start >= %d", startTime.UnixMilli()))
+	endTime, _ := parseDateFromArg(true, end, now)
+	filters = append(filters, fmt.Sprintf("end <= %d", endTime.UnixMilli()))
+
+	if endTime.Before(startTime) {
+		return "", fmt.Errorf("end is before start")
+	}
+
+	query := fmt.Sprintf("select * from reps where %s order by start desc;", strings.Join(filters, " and "))
+
+	return query, nil
+}
+
+func queryToReps(query string) (reps []Rep, err error) {
+	statsDb, err := SweetDb()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database: %s\n", err)
+	}
+
+	reps, err = GetReps(statsDb, query)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reps: %s\n", err)
+	}
+
+	return
+}
+
+func argsToColumnFilter(cmd *cobra.Command) []string {
+	cols := []string{"start"}
+	name := cmd.Flag("name").Value.String()
+	showName := name == "" || strings.Contains(name, "*")
+	if showName {
+		cols = append(cols, "name")
+	}
+	possibleCols := []string{"wpm", "raw", "acc", "errs", "miss", "dur"}
```

Comment:
Should I create constants for all of the column names in the database? It would be nice to avoid misnaming things in the future.